cli/cook/subcommands/show.py

```diff
@@ -0,0 +1,343 @@
+            ['Slave', instance['slave_id']],
+            ['Job', '%s (%s)' % (job['name'], job['uuid'])]]
+    if len(instance['ports']) > 0:
+        left.append(['Ports Allocated', format_list(instance['ports'])])
+
+    right = [['Run Time', format_instance_run_time(instance)],
+             ['Instance Status', format_instance_status(instance)],
+             ['Job Status', format_job_status(job)]]
+    if 'exit_code' in instance:
+        right.append(['Exit Code', instance['exit_code']])
+
+    left_table = tabulate(left, tablefmt='plain')
+    right_table = tabulate(right, tablefmt='plain')
+    instance_tables = juxtapose_text(left_table, right_table, 15)
+    return '\n=== Job Instance: %s ===\n\n%s' % (instance['task_id'], instance_tables)
+
+
+def tabulate_group(cluster_name, group):
+    """Given a group, returns a string containing a table for the group fields"""
+    left = [['Cluster', cluster_name]]
+    if group['host_placement']['type'] == 'all':
+        left.append(['Host Placement', 'all hosts'])
+    else:
+        left.append(['Host Placement', format_dict(group['host_placement'])])
+    if group['straggler_handling']['type'] == 'none':
+        left.append(['Straggler Handling', 'none'])
+    else:
+        left.append(['Straggler Handling', format_dict(group['straggler_handling'])])
+
+    right = [['# Completed', group['completed']],
+             ['# Running', group['running']],
+             ['# Waiting', group['waiting']]]
+
+    num_jobs = len(group['jobs'])
+    jobs = 'Job group contains %s job%s:\n%s' % (num_jobs, '' if num_jobs == 1 else 's', '\n'.join(group['jobs']))
+
+    left_table = tabulate(left, tablefmt='plain')
+    right_table = tabulate(right, tablefmt='plain')
+    group_tables = juxtapose_text(left_table, right_table, 15)
+    return '\n=== Job Group: %s (%s) ===\n\n%s\n\n%s' % (group['uuid'], group['name'], group_tables, jobs)
+
+
+def show_data(cluster_name, data, format_fn):
+    """Iterates over the data collection and formats and prints each element"""
+    count = len(data)
+    if count > 0:
+        tables = [format_fn(cluster_name, datum) for datum in data]
+        output = '\n\n'.join(tables)
+        print(output)
+        print()
+    return count
+
+
+def query_cluster(cluster, uuids, pred, timeout, interval, make_request_fn):
+    """
+    Queries the given cluster for the given uuids with
+    an optional predicaite, pred, that must be satisfied
+    """
+
+    def satisfy_pred(cluster_, uuids_):
+        resp_ = make_request_fn(cluster_, uuids_)
+        return pred(resp_.json())
+
+    try:
+        resp = make_request_fn(cluster, uuids)
+        if resp.status_code == requests.codes.ok:
+            entities = resp.json()
+            if pred and not pred(entities):
+                entities = wait_until(lambda: satisfy_pred(cluster, uuids), timeout, interval)
+                if not entities:
+                    raise Exception('Timeout waiting for response')
```

Comment:
When I run `cs wait`, it times out every 30 seconds. The request code is shared between `show` and `wait`, so I can understand why `show` has a timeout, but maybe you should override the behavior for `wait` to retry indefinitely. (If I want to cancel a wait, I'm going to want to explicitly send it a signal via `^c` or `kill`.)

Reply:
I changed it so that `wait` now waits indefinitely by default, but still supports the optional `--timeout` flag

Reply:
:+1: I like the added flexibility of having the timeout flag too.